torchvision/models/_api.py

```diff
@@ -199,7 +196,7 @@ def list_models(module: Optional[ModuleType] = None) -> List[str]:
     return sorted(models)
 
 
-def get_model_builder(name: str) -> Callable[..., M]:
+def get_model_builder(name: str) -> Callable[..., nn.Module]:
```

Comment:
How of curiosity why do we need to change this one, and not all the other usages of `M` e.g. in `register_model()` above?

Reply:
`TypeVar` is used to create a "link" between parameters. For example I can do something like

```py
def do_nothing(foo: M) -> M:
    return foo
```

This tells `mypy` "whatever we have as input type, we return as output type as well". This is useful in case the function can handle multiple types, e.g.

```py
T = TypeVar("T", int, float)

def add(a: T, b: T) -> T:
    return a + b
``` 

In the case above, there is no input with the type `M` and so `mypy` complains that it can't figure out the `M` in the output. 

Correct me if I'm wrong @datumbox, but I guess the annotation was intended to mean "we return any kind of model" here. Since `M` is defined as 

https://github.com/pytorch/vision/blob/06d9726d66f97ed8e3d3910da246a52be9c6e168/torchvision/models/_api.py#L168

we can simply use `nn.Module` here.